# Data Product Arquitecture - Food Inspections Project

Team:

* Jóse Roberto Pérez - 173013
* Santiago Battezzati - 197727
* Carlos Geovany Pérez - 197614
* Mariana Lugo - 118626

## Imports

#### Standard library imports

In [1]:
import os
import sys
from datetime import datetime

#### Third party imports

In [2]:
import pandas as pd
from pandas_profiling import ProfileReport
pd.options.display.max_columns = 25

import seaborn as sns

#### Local application imports

In [3]:
sys.path.append("../")

In [21]:
%reload_ext autoreload
%autoreload 2

from src.utils.utils import *

---

## Importing and cleaning data
Food Inspections in Chicago: https://data.cityofchicago.org/Health-Human-Services/Food-Inspections/4ijn-s7e5

In [36]:
df = pd.read_csv('../data/Food_Inspections.csv')
df.head()

Inspection ID            DBA Name          AKA Name  License #  \
0        2453552  CLAMP DOWN BURGERS               NaN  2749943.0   
1        2386633              JIN JU            JIN JU    27137.0   
2        2386595       LA BIZNAGA #2     LA BIZNAGA #2  2708992.0   
3        2386464    Uni Sushi Bristo  Uni Sushi Bistro  2262637.0   
4        2386398       KIKI'S BISTRO     KIKI'S BISTRO    22899.0   

  Facility Type             Risk              Address     City State      Zip  \
0           NaN  Risk 2 (Medium)  1742 W DIVISION ST   CHICAGO    IL  60622.0   
1    Restaurant    Risk 1 (High)     5203 N CLARK ST   CHICAGO    IL  60640.0   
2           NaN    Risk 1 (High)  2949 W BELMONT AVE   CHICAGO    IL  60618.0   
3    Restaurant    Risk 1 (High)    1752 W NORTH AVE   CHICAGO    IL  60622.0   
4    Restaurant    Risk 1 (High)   900 N FRANKLIN ST   CHICAGO    IL  60610.0   

  Inspection Date Inspection Type   Results Violations   Latitude  Longitude  \
0      10/20/2020         License  No Entry        NaN  41.903387 -87.671740   
1      08/28/2020         Canvass  No Entry        NaN  41.976301 -87.668276   
2      08/27/2020       Complaint  No Entry        NaN  41.939256 -87.702270   
3      08/25/2020       Complaint  No Entry        NaN  41.910676 -87.672205   
4      08/24/2020         Canvass  No Entry        NaN  41.898998 -87.635921   

                                   Location  
0  (-87.67174026586648, 41.903386755553484)  
1   (-87.66827593789948, 41.97630115368914)  
2  (-87.70226967930802, 41.939255926667535)  
3   (-87.67220465807979, 41.91067561170382)  
4   (-87.63592067312285, 41.89899799424835)

Limpieza de nombres de las columnas:

In [37]:
clean_col_names(df)

inspection_id                dba_name             aka_name  license_#  \
0             2453552      CLAMP DOWN BURGERS                  NaN  2749943.0   
1             2386633                  JIN JU               JIN JU    27137.0   
2             2386595           LA BIZNAGA #2        LA BIZNAGA #2  2708992.0   
3             2386464        Uni Sushi Bristo     Uni Sushi Bistro  2262637.0   
4             2386398           KIKI'S BISTRO        KIKI'S BISTRO    22899.0   
...               ...                     ...                  ...        ...   
215062         229233    ZULLO'S MARKETS, LLC       ZULLO'S MARKET  2016915.0   
215063          68205   LA FONDA CHIQUITA TLC    LA FONDA CHIQUITA  2017215.0   
215064          88234         KABAB CUISINE 2      KABAB CUISINE 2  2014076.0   
215065          74311  MARGARITA DISTRIBUTORS                  NaN  2017298.0   
215066          67838     OUR LADY OF LOURDES  OUR LADY OF LOURDES    42577.0   

                  facility_type             risk  \
0                           NaN  Risk 2 (Medium)   
1                    Restaurant    Risk 1 (High)   
2                           NaN    Risk 1 (High)   
3                    Restaurant    Risk 1 (High)   
4                    Restaurant    Risk 1 (High)   
...                         ...              ...   
215062               Restaurant  Risk 2 (Medium)   
215063               Restaurant    Risk 1 (High)   
215064               Restaurant    Risk 1 (High)   
215065                Wholesale     Risk 3 (Low)   
215066  Daycare (Under 2 Years)    Risk 1 (High)   

                             address     city state      zip inspection_date  \
0                1742 W DIVISION ST   CHICAGO    IL  60622.0      10/20/2020   
1                   5203 N CLARK ST   CHICAGO    IL  60640.0      08/28/2020   
2                2949 W BELMONT AVE   CHICAGO    IL  60618.0      08/27/2020   
3                  1752 W NORTH AVE   CHICAGO    IL  60622.0      08/25/2020   
4                 900 N FRANKLIN ST   CHICAGO    IL  60610.0      08/24/2020   
...                              ...      ...   ...      ...             ...   
215062             131 N Clinton ST   CHICAGO    IL  60661.0      02/18/2010   
215063          5940 W DIVERSEY AVE   CHICAGO    IL  60639.0      02/09/2010   
215064           6320 N LINCOLN AVE   CHICAGO    IL  60659.0      02/01/2010   
215065  2332 S BLUE ISLAND AVE BLDG   CHICAGO    IL  60608.0      02/19/2010   
215066            1449 S KEELER AVE   CHICAGO    IL  60623.0      01/12/2010   

              inspection_type   results  \
0                     License  No Entry   
1                     Canvass  No Entry   
2                   Complaint  No Entry   
3                   Complaint  No Entry   
4                     Canvass  No Entry   
...                       ...       ...   
215062                License      Fail   
215063                License      Fail   
215064  License Re-Inspection      Pass   
215065  License Re-Inspection      Pass   
215066                Canvass      Fail   

                                               violations   latitude  \
0                                                     NaN  41.903387   
1                                                     NaN  41.976301   
2                                                     NaN  41.939256   
3                                                     NaN  41.910676   
4                                                     NaN  41.898998   
...                                                   ...        ...   
215062                                                NaN  41.884188   
215063  12. HAND WASHING FACILITIES: WITH SOAP AND SAN...  41.931260   
215064                                                NaN  41.996139   
215065                                                NaN  41.849940   
215066  18. NO EVIDENCE OF RODENT OR INSECT OUTER OPEN...  41.861003   

        longitude                                  location  
0    

**Número de variables:**

In [38]:
count_vars(df)

Número de variables en los datos --> 17


**Número de observaciones:**

In [39]:
count_obs(df)

Número de observaciones en los datos --> 215067


**¿Qué variables son?**

In [40]:
count_unique_obs(df)

inspection_id      215067
dba_name            28748
aka_name            27355
license_#           39103
facility_type         500
risk                    4
address             18522
city                   70
state                   4
zip                   112
inspection_date      2796
inspection_type       110
results                 7
violations         156694
latitude            17246
longitude           17246
location            17246
dtype: int64

In [41]:
cat_vars=['dba_name','aka_name','facility_type','risk','address','city','state','inspection_type',
          'results','violations']
data_profiling_categ(df, cat_vars)

*********************************
Variable Categorica dba_name
*********************************


Info                            dba_name
Num_Registros                     215067
Num_de_categorias                  28748
Moda                              SUBWAY
Valores_faltantes                      0
Top1                      [SUBWAY, 2905]
Top2               [DUNKIN DONUTS, 1578]
Top3                   [MCDONALD'S, 614]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                             
SUBWAY                                 2905       1.4%
DUNKIN DONUTS                          1578       0.7%
MCDONALD'S                              614       0.3%
7-ELEVEN                                469       0.2%
CHIPOTLE MEXICAN GRILL                  344       0.2%
...                                     ...        ...
SUGAR                                     1       0.0%
BIRDIE PAK PRODUCTS INC                   1       0.0%
KING FISH & WING                          1       0.0%
TACOS REYNA                               1       0.0%
LATERMINAL JESUS MINI MARKET              1       0.0%

[28748 rows x 2 columns]

None



*********************************
Variable Categorica aka_name
*********************************


Info                            aka_name
Num_Registros                     215067
Num_de_categorias                  27355
Moda                              SUBWAY
Valores_faltantes                   2486
Top1                      [SUBWAY, 3565]
Top2               [DUNKIN DONUTS, 1586]
Top3                     [7-ELEVEN, 882]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                          
SUBWAY                              3565       1.7%
DUNKIN DONUTS                       1586       0.7%
7-ELEVEN                             882       0.4%
MCDONALD'S                           834       0.4%
BURGER KING                          448       0.2%
...                                  ...        ...
WALGREENS #807                         1       0.0%
CLOUD COOKIE                           1       0.0%
CORDI-MARIAN CENTER                    1       0.0%
DONG WON HERBS                         1       0.0%
BEATRICE LIQUOR & GROCERY              1       0.0%

[27355 rows x 2 columns]

None



*********************************
Variable Categorica facility_type
*********************************


Info                        facility_type
Num_Registros                      215067
Num_de_categorias                     500
Moda                           Restaurant
Valores_faltantes                    4889
Top1                 [Restaurant, 143207]
Top2               [Grocery Store, 28084]
Top3                      [School, 13206]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                             
Restaurant                           143207      66.6%
Grocery Store                         28084      13.1%
School                                13206       6.1%
Children's Services Facility           3707       1.7%
Bakery                                 3153       1.5%
...                                     ...        ...
GAS STATION FOOD STORE                    1       0.0%
LAUNDROMAT                                1       0.0%
GROCERY & RESTAURANT                      1       0.0%
PROTEIN SHAKE BAR                         1       0.0%
TAVERN/LIQUOR                             1       0.0%

[500 rows x 2 columns]

None



*********************************
Variable Categorica risk
*********************************


Info                                   risk
Num_Registros                        215067
Num_de_categorias                         4
Moda                          Risk 1 (High)
Valores_faltantes                        69
Top1                [Risk 1 (High), 153815]
Top2               [Risk 2 (Medium), 41971]
Top3                  [Risk 3 (Low), 19184]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                
Risk 1 (High)           153815      71.5%
Risk 2 (Medium)          41971      19.5%
Risk 3 (Low)             19184       8.9%
All                         28       0.0%

None



*********************************
Variable Categorica address
*********************************


Info                                  address
Num_Registros                          215067
Num_de_categorias                       18522
Moda                       11601 W TOUHY AVE 
Valores_faltantes                           0
Top1               [11601 W TOUHY AVE , 2674]
Top2                [5700 S CICERO AVE , 593]
Top3                 [500 W MADISON ST , 465]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                       
11601 W TOUHY AVE                2674       1.2%
5700 S CICERO AVE                 593       0.3%
500 W MADISON ST                  465       0.2%
2300 S THROOP ST                  410       0.2%
324 N LEAVITT ST                  391       0.2%
...                               ...        ...
1811 W HARRISON ST                  1       0.0%
1313 W MORSE AVE                    1       0.0%
5300 N DAMEN AVE                    1       0.0%
4221 W IRVING PARK RD               1       0.0%
1958 W NORTH AVE                    1       0.0%

[18522 rows x 2 columns]

None



*********************************
Variable Categorica city
*********************************


Info                            city
Num_Registros                 215067
Num_de_categorias                 70
Moda                         CHICAGO
Valores_faltantes                158
Top1               [CHICAGO, 214166]
Top2                  [Chicago, 345]
Top3                  [chicago, 111]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                           
CHICAGO            214166      99.6%
Chicago               345       0.2%
chicago               111       0.1%
CCHICAGO               48       0.0%
SCHAUMBURG             27       0.0%
...                   ...        ...
WESTMONT                1       0.0%
LAKE BLUFF              1       0.0%
PALOS PARK              1       0.0%
WHEATON                 1       0.0%
OAK LAWN                1       0.0%

[70 rows x 2 columns]

None



*********************************
Variable Categorica state
*********************************


Info                      state
Num_Registros            215067
Num_de_categorias             4
Moda                         IL
Valores_faltantes            50
Top1               [IL, 215014]
Top2                    [NY, 1]
Top3                    [IN, 1]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                          
IL                215014     100.0%
NY                     1       0.0%
IN                     1       0.0%
WI                     1       0.0%

None



*********************************
Variable Categorica inspection_type
*********************************


Info                              inspection_type
Num_Registros                              215067
Num_de_categorias                             110
Moda                                      Canvass
Valores_faltantes                               1
Top1                            [Canvass, 114224]
Top2                             [License, 28292]
Top3               [Canvass Re-Inspection, 22824]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                                  
Canvass                                   114224      53.1%
License                                    28292      13.2%
Canvass Re-Inspection                      22824      10.6%
Complaint                                  19908       9.3%
License Re-Inspection                       9631       4.5%
...                                          ...        ...
citation re-issued                             1       0.0%
CANVASS RE INSPECTION OF CLOSE UP              1       0.0%
CANVASS SPECIAL EVENTS                         1       0.0%
CANVAS                                         1       0.0%
CANVASS FOR RIB FEST                           1       0.0%

[110 rows x 2 columns]

None



*********************************
Variable Categorica results
*********************************


Info                                   results
Num_Registros                           215067
Num_de_categorias                            7
Moda                                      Pass
Valores_faltantes                            0
Top1                            [Pass, 112619]
Top2                             [Fail, 41511]
Top3               [Pass w/ Conditions, 32540]

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                     
Pass                         112619      52.4%
Fail                          41511      19.3%
Pass w/ Conditions            32540      15.1%
Out of Business               18706       8.7%
No Entry                       7332       3.4%
Not Ready                      2287       1.1%
Business Not Located             72       0.0%

None



*********************************
Variable Categorica violations
*********************************


Info                                                      violations
Num_Registros                                                 215067
Num_de_categorias                                             156694
Moda               32. FOOD AND NON-FOOD CONTACT SURFACES PROPERL...
Valores_faltantes                                              57424
Top1               [32. FOOD AND NON-FOOD CONTACT SURFACES PROPER...
Top2               [30. FOOD IN ORIGINAL CONTAINER, PROPERLY LABE...
Top3               [45. FOOD HANDLER REQUIREMENTS MET - Comments:...

None
Valores de las categorias y sus proporciones


Observaciones proporción
Categoría                                                                   
32. FOOD AND NON-FOOD CONTACT SURFACES PROPERLY...             11       0.0%
30. FOOD IN ORIGINAL CONTAINER, PROPERLY LABELE...             10       0.0%
45. FOOD HANDLER REQUIREMENTS MET - Comments: A...             10       0.0%
3. MANAGEMENT, FOOD EMPLOYEE AND CONDITIONAL EM...              7       0.0%
45. FOOD HANDLER REQUIREMENTS MET - Comments: A...              7       0.0%
...                                                           ...        ...
5. PROCEDURES FOR RESPONDING TO VOMITING AND DI...              1       0.0%
57. ALL FOOD EMPLOYEES HAVE FOOD HANDLER TRAINI...              1       0.0%
18. NO EVIDENCE OF RODENT OR INSECT OUTER OPENI...              1       0.0%
29. PREVIOUS MINOR VIOLATION(S) CORRECTED 7-42-...              1       0.0%
34. FLOORS: CONSTRUCTED PER CODE, CLEANED, GOOD...              1       0.0%

[156694 rows x 2 columns]

None





### Análisis de texto
Issue #13 GitHub

#### 1. Limpieza de campo `facility_type`

In [43]:
print(len(df["facility_type"].unique()))
df["facility_type"].unique()

501


array([nan, 'Restaurant', "Children's Services Facility", 'Grocery Store',
       'GAS STATION', 'Bakery', 'Long Term Care', 'Mobile Food Preparer',
       'School', 'Golden Diner', 'HOSTEL', 'school cafeteria',
       'Daycare (2 - 6 Years)', 'Daycare Above and Under 2 Years',
       'Hospital', 'Pop-Up Establishment Host-Tier II', 'Catering',
       'BREWERY', 'Liquor', 'TAVERN', 'Daycare Combo 1586',
       'Mobile Food Dispenser', 'SCHOOL', 'VENDING COMMISSARY',
       'Shared Kitchen User (Long Term)', 'COFFEE SHOP', 'Shared Kitchen',
       'LIQUOR/COFFEE KIOSK', 'SUPPORTIVE LIVING',
       'Daycare (Under 2 Years)', 'Mobile Prepared Food Vendor',
       'meat packing', 'CHARTER SCHOOL', 'PRIVATE SCHOOL',
       'Mobile Frozen Desserts Vendor', 'GROCERY STORE/COOKING SCHOOL',
       'CHURCH (SPECIAL EVENTS)', 'Special Event', 'STADIUM', 'Wholesale',
       'RESTAURANT/GROCERY STORE', 'bar', 'DISTRIBUTION CENTER',
       'Wrigley Roof Top', 'DAYCARE', 'PREPACAKAGED FOODS',
       

In [56]:
df["facility_type"].fillna("nan", inplace=True)

In [57]:
df["facility_type"] = df["facility_type"].apply(lambda x: clean_txt(x))

In [60]:
print(len(df["facility_type"].unique()))
df["facility_type"].value_counts()

448


restaurant                      143207
grocery_store                    28088
school                           13219
nan                               4889
children-s_services_facility      3707
                                 ...  
grocery-liquor                       1
religious                            1
mobil_food_1315                      1
kids_cafe-                           1
health_club                          1
Name: facility_type, Length: 448, dtype: int64

In [63]:
dfx = df["facility_type"].value_counts().to_frame()
dfx

facility_type
restaurant                           143207
grocery_store                         28088
school                                13219
nan                                    4889
children-s_services_facility           3707
...                                     ...
grocery-liquor                            1
religious                                 1
mobil_food_1315                           1
kids_cafe-                                1
health_club                               1

[448 rows x 1 columns]

In [69]:
dfx["facility_type"].value_counts()

2        66
1        63
3        59
4        30
5        26
         ..
57        1
63        1
65        1
66        1
28088     1
Name: facility_type, Length: 78, dtype: int64

In [78]:
mr1 = dfx["facility_type"] <= 1
mr2 = dfx.index.str.contains("rest")
dfx.loc[(mr2), :][:50]

facility_type
restaurant                            143207
grocery-restaurant                        57
restaurant-bar                            41
grocery_-_restaurant                      29
restaurant-grocery_store                  27
gas_station-restaurant                    15
grocery-_restaurant                       13
restaurant-hospital                       11
restaurant.banquet_halls                   9
rest-grocery                               9
grocery_store-_restaurant                  9
restaurant-protein_shake_bar-              9
restuarant_and_bar                         9
theater_-_restaurant                       8
rest-rooftop                               8
restaurant-bakery                          7
rest-gym                                   7
restaurant-grocery                         4
commisary_restaurant                       3
tavern-restaurant                          3
bakery-restaurant                          2
restaurant-bar-theater                     2
bakery-_restaurant                         2
restaurant-liquor                          2
restaurant_and_liquor                      1

In [59]:
mr1 = df["facility_type"].str.contains("day")
df.loc[mr1, ["facility_type"]]

facility_type
146               daycare_-2_-_6_years-
148     daycare_above_and_under_2_years
156               daycare_-2_-_6_years-
217               daycare_-2_-_6_years-
240     daycare_above_and_under_2_years
...                                 ...
214998            daycare_-2_-_6_years-
214999            daycare_-2_-_6_years-
215023            daycare_-2_-_6_years-
215045            daycare_-2_-_6_years-
215066          daycare_-under_2_years-

[6653 rows x 1 columns]

**Notas de análisis:**
- @9/Feb/2020
    - Una vez limpiados los textos, la cantidad de únicos pasó de 501 a 448.
    - Tal vez podríamos agrupar las categorías por aparición de palabras clave (e.g. `daycare` OR `day_care`).
    - Todas las que caigan fuera de las categorías, serán incluidas en una reserva llamada "Otros"
    - Podría valer la pena hacer un análisis de clústering para agrupar palabras por similitudes y así definir las categorías. (tal vez va a ser difícil hacer estas asociaciones.)

#### 2. Limpieza de campo `inspection_date`

#### 3. Limpieza de campo `violations`

---

## 1. ¿Cuáles son los tipos 10 de establecimientos que más violaciones cometen? (Facility Type)


In [ ]:
conteos_ft=df.groupby(["results","facility_type"]).size().reset_index(name="counts")
conteos_ft=conteos_ft.pivot("facility_type","results","counts")
conteos_ft=conteos_ft.fillna(0)
conteos_ft['Total']=conteos_ft.sum(1)
conteos_ft =conteos_ft.sort_values('Fail',ascending=False)
conteos_ft

In [ ]:
n = 10
top10=df['facility_type'].value_counts()[:10].sort_values(ascending=False)
top10

Se muestra el porcentaje del tipo de establecimiento acuerdo con el resultado de la inspección:

In [ ]:
proporciones_ft=round(conteos_ft.iloc[:,0:7].apply(lambda x: x/x.sum()*100, axis=1),2)
proporciones_ft=proporciones_ft.sort_values('Fail',ascending=False)
proporciones_ft

## 2. ¿Qué tipo de inspección detecta más establecimientos fuera de regla? (Inspection Type)

In [ ]:
conteos_it=df.groupby(["results","inspection_type"]).size().reset_index(name="counts")
conteos_it=conteos_it.pivot("inspection_type","results","counts")
conteos_it=conteos_it.fillna(0)
conteos_it['Total']=conteos_it.sum(1)
conteos_it =conteos_it.sort_values('Fail',ascending=False)
conteos_it

Se muestra el porcentaje del tipo de inspección de acuerdo con los resultados:

In [ ]:
proporciones_it=round(conteos_it.iloc[:,0:7].apply(lambda x: x/x.sum()*100, axis=1),2)
proporciones_it=proporciones_it.sort_values('Fail',ascending=False)
proporciones_it

Se muestra el top 20 de tipos de inspección con un porcentaje mayor de inspecciones no reprobadas.

In [ ]:
top20=proporciones_it.iloc[0:19,:]
top20

## 3. Relación entre el riesgo y resultado, ¿los más riesgosos no pasan la insepección?

In [ ]:
rr=df.groupby(['results','risk']).size().reset_index(name="counts")
rr=rr.pivot("results", "risk","counts")
rr=rr.fillna(0)
rr['Total']=rr.sum(1)
rr =rr.sort_values('Risk 1 (High)',ascending=False)
rr

In [ ]:
prop=round(rr.iloc[:,0:4].apply(lambda x: x/x.sum()*100, axis=1),2)
prop

In [ ]:
prop.plot(kind='bar',stacked=True)

## 4. ¿Cómo es el comportamiento de las inspecciones en el tiempo?

Se separa inspection_date en tres diferentes columnas: day, month, year y las agrega al dataframe:

In [ ]:
fecha= df.inspection_date.str.split("/", n=2,expand=True)
df['day']=fecha[1]
df['month']=fecha[0]
df['year']=fecha[2]

In [ ]:
df['inspection_date'] = pd.to_datetime(df['inspection_date'])

In [ ]:
# Columnas de year, month y counts
fecha_separated = df.groupby(['year','month']).size().reset_index(name="conteos")
# Hacemos tabla donde las columnas son años y las filas mes 
fecha_separated_p = fecha_separated.pivot("month", "year","conteos")
# Graficamos 
sns.set(style='whitegrid',rc={'figure.figsize':(10,10)},palette='Set3')
h =sns.lineplot(data=fecha_separated_p,linewidth = 3,dashes=False)
plt.legend(fontsize='x-large', title_fontsize='40',loc='lower right')
h.set(xlabel='Meses',ylabel='Conteos de Inspecciones')
plt.show()

## 5. ¿Cómo es el comportamiento por tipo de resultados en el tiempo?

In [ ]:
fecha_separated2 = df.groupby(['year','month','results']).size().reset_index(name="counts")
fecha_separated2['date']= fecha_separated2['year']+'-'+fecha_separated2['month']
g=fecha_separated2.drop(['year','month'],1)
g=g.set_index('date')
sns.set(style='whitegrid',rc={'figure.figsize':(20,10)},palette='Set2')
sns.lineplot(x='date',y='counts',hue='results',data=g,linewidth = 2)
plt.xticks(rotation= 90,size=8)
plt.show()

Se muestra el comportamiento anterior en porcentaje de inspecciones:

In [ ]:
g2=fecha_separated2.drop(['year','month'],1)
g2=g2.pivot("date", "results","counts")
g2=g2.fillna(0)
g2['total']=g2.sum(1)
prop2=round(g2.iloc[:,0:5].apply(lambda x: x/x.sum()*100, axis=1),2)
sns.set(style='whitegrid',rc={'figure.figsize':(20,10)},palette='Set2')
h =sns.lineplot(data=prop2,linewidth = 2,dashes=False,marker='o')
plt.xticks(rotation= 90)
plt.legend(fontsize='x-large', title_fontsize='40')
plt.show()

## 6. ¿Es importante la temporalidad de año, mes y día para las inspecciones?

Con variable inspection_date en formato tiempo creamos conteos del número de inspecciones hechas por día:

In [ ]:
conteos_fechas = df.groupby(['inspection_date']).size().reset_index(name='conteos')
# hacemos que inspection_date sea el identificador de las filas
conteos_fechas = conteos_fechas.set_index('inspection_date')


In [ ]:
# Separamos los días, meses, años
conteos_fechas.insert(1, 'year', conteos_fechas.index.year)
conteos_fechas.insert(2, 'month', conteos_fechas.index.month)
conteos_fechas.insert(3, 'day', conteos_fechas.index.day)
conteos_fechas = conteos_fechas.copy()

In [ ]:
# Graficamos
fig, axs = plt.subplots(figsize=(30,20),ncols=6, nrows=2)
heatmaps(conteos_fechas,2010,axs[0][0],'no','Day')
heatmaps(conteos_fechas,2011,axs[0][1],'no','')
heatmaps(conteos_fechas,2012,axs[0][2],'no','')
heatmaps(conteos_fechas,2013,axs[0][3],'no','')
heatmaps(conteos_fechas,2014,axs[0][4],'no','')
heatmaps(conteos_fechas,2015,axs[0][5],'yes','')
heatmaps(conteos_fechas,2016,axs[1][0],'no','')
heatmaps(conteos_fechas,2017,axs[1][1],'no','')
heatmaps(conteos_fechas,2018,axs[1][2],'no','')
heatmaps(conteos_fechas,2019,axs[1][3],'no','')
heatmaps(conteos_fechas,2020,axs[1][4],'no','')
heatmaps(conteos_fechas,2021,axs[1][5],'yes','')

**Vemos si para si existe un patrón temporal en el universo de inspecciones que resultaron como reprobadas (Results="Fail"):**

In [ ]:
fail=df[df.results=="Fail"]
len(fail)

In [ ]:
conteos_f = fail.groupby(['inspection_date']).size().reset_index(name='conteos')
conteos_f= conteos_f.set_index('inspection_date')
conteos_f.insert(1, 'year', conteos_f.index.year)
conteos_f.insert(2, 'month', conteos_f.index.month)
conteos_f.insert(3, 'day', conteos_f.index.day)
conteos_f = conteos_f.copy()

In [ ]:
fig, axs = plt.subplots(figsize=(30,20),ncols=6, nrows=2)
heatmaps(conteos_f,2010,axs[0][0],'no','Day')
heatmaps(conteos_f,2011,axs[0][1],'no','')
heatmaps(conteos_f,2012,axs[0][2],'no','')
heatmaps(conteos_f,2013,axs[0][3],'no','')
heatmaps(conteos_f,2014,axs[0][4],'no','')
heatmaps(conteos_f,2015,axs[0][5],'yes','')
heatmaps(conteos_f,2016,axs[1][0],'no','Day')
heatmaps(conteos_f,2017,axs[1][1],'no','')
heatmaps(conteos_f,2018,axs[1][2],'no','')
heatmaps(conteos_f,2019,axs[1][3],'no','')
heatmaps(conteos_f,2020,axs[1][4],'yes','')
heatmaps(conteos_f,2021,axs[1][5],'yes','')

## 7. Mapa

In [ ]:
sns.set(style='whitegrid',rc={'figure.figsize':(20,10)},palette='Set1')
s = sns.lmplot(x='longitude',y='latitude',data=df,hue='results',col='results',fit_reg=False,height=6,
               scatter_kws={"s":1,"alpha":.5})

In [ ]:
cols = ['Results','Latitude','Longitude']
rest = pd.read_csv('../data/Food_Inspections.csv', usecols=cols)

Eliminamos los registros sin coordenadas completas:

In [ ]:
rest = rest.dropna(subset=['Latitude'])
rest = rest.dropna(subset=['Longitude'])

Filtramos únicamente aquellos registros que pasaron o no pasaron la prueba

In [ ]:
passed = rest[rest['Results']=='Pass']
failed = rest[rest['Results']=='Fail']

In [ ]:
len(passed), len(failed)

Imagen de la ciudad de chicago:

In [ ]:
chicago = plt.imread('../images/chicago.png')

In [ ]:
boundires = ((rest.Longitude.min(),   rest.Longitude.max(),      
         rest.Latitude.min(), rest.Latitude.max()))

In [ ]:
fig, ax = plt.subplots(1,2, sharex=True, sharey=True, figsize = (20,20))

ax[0].scatter(passed.Longitude, passed.Latitude, zorder=2, alpha=0.1, c='b', s=2)
ax[0].imshow(chicago, zorder=0, extent=boundires, aspect='equal')
ax[0].title.set_text('Results: Pass')

ax[1].scatter(failed.Longitude, failed.Latitude, zorder=2, alpha=0.1, c='r', s=2)
ax[1].imshow(chicago, zorder=0, extent=boundires, aspect='equal')
ax[1].title.set_text('Results: Fail')

plt.show()

---
---